pkg/resources/external_volume.go

```diff
@@ -379,17 +377,6 @@ func UpdateContextExternalVolume(ctx context.Context, d *schema.ResourceData, me
 			// would otherwise be necessary as a minimum of 1 storage location per external volume is required.
 			// The alternative solution of adding volumes before removing them isn't possible as
 			// name must be unique for storage locations
-			if len(removedLocations) > 1 {
-				// To ensure the name of the temporary storage location is unique,
-				// first remove all but 1 storage locations from the list
-				for _, removedStorageLocation := range removedLocations[1:] {
-					err = removeStorageLocation(removedStorageLocation, client, ctx, id)
-					if err != nil {
-						return diag.FromErr(err)
-					}
-				}
-			}
-
 			temp_storage_location, err := CopyStorageLocationWithTempName(removedLocations[0])
 			if err != nil {
 				return diag.FromErr(err)
```

Comment:
I think that `CopyStorageLocationWithTempName` is not suitable here - we can create an arbitrary storage location with our special name. We can leave it as is, but rename the function to `prepareSentinelRequest` or something similar.

Reply:
That's also an option - we could have a storage location with placeholder values for the storage location, arn etc. I think this would work as it works in the tests, so it seems Snowflake doesn't try to verify access to the underlying data. This happens when creating an Iceberg table.

I stuck with this though as not having storage locations with placeholder values seemed better long term, for example if Snowflake does implement verifications here in the future. Not sure either way - happy to go with whatever think is best.

Reply:
I'll update the name of the function to `CopySentinelStorageLocation` for now